crates/blockchain/vm.rs

```diff
@@ -32,13 +33,23 @@ impl VmDatabase for StoreVmDatabase {
     }
 
     fn get_block_hash(&self, block_number: u64) -> Result<H256, EvmError> {
```

Comment:
Most of the time we will be executing blocks that belong to the canonical chain (aka the ones we get from `forkChoiceUpdate`, `newPayload` & syncing). Wouldn't it be better to first check if the block hash we have is canonical, and if it is then we can use  `get_canonical_block_hash` as the ancestor block will also be canonical. This will save us from having to go through all ancestors on our most relevant use cases

Reply:
As discussed offline, created an issue for this as there's an async API requirement that needs more discussion: #2990